sql: TestErrorOnCommitFinalizesTxn failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/d6a4a0d6a7e84debc0195822f6cd196405c584c4

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestErrorOnCommitFinalizesTxn PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=858423&tab=buildLog

```
=== RUN   TestErrorOnCommitFinalizesTxn
W180824 05:59:34.129435 176306 server/status/runtime.go:294  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180824 05:59:34.224368 176306 server/server.go:830  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180824 05:59:34.225163 176306 base/addr_validation.go:260  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180824 05:59:34.225276 176306 base/addr_validation.go:300  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180824 05:59:34.253011 176306 server/config.go:496  [n?] 1 storage engine initialized
I180824 05:59:34.253261 176306 server/config.go:499  [n?] RocksDB cache size: 128 MiB
I180824 05:59:34.253304 176306 server/config.go:499  [n?] store 0: in-memory, size 0 B
I180824 05:59:34.492547 176306 server/node.go:373  [n?] **** cluster 778669e8-bbea-4418-8b9c-cc57e630d855 has been created
I180824 05:59:34.492824 176306 server/server.go:1401  [n?] **** add additional nodes by specifying --join=127.0.0.1:38525
I180824 05:59:34.494499 176306 gossip/gossip.go:382  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38525" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:12 > build_tag:"v2.1.0-alpha.20180702-1973-gd6a4a0d" started_at:1535090374493993901 
I180824 05:59:34.520530 176306 storage/store.go:1541  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180824 05:59:34.554931 176306 server/node.go:476  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=1, queries=0.00, writes=0.00, bytesPerReplica={p10=7103.00 p25=7103.00 p50=7103.00 p75=7103.00 p90=7103.00 pMax=7103.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180824 05:59:34.555811 176306 storage/stores.go:242  [n1] read 0 node addresses from persistent storage
I180824 05:59:34.556628 176306 server/node.go:697  [n1] connecting to gossip network to verify cluster ID...
I180824 05:59:34.562775 176306 server/node.go:722  [n1] node connected via gossip and verified as part of cluster "778669e8-bbea-4418-8b9c-cc57e630d855"
I180824 05:59:34.573922 176306 server/node.go:546  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180824 05:59:34.585446 176306 server/status/recorder.go:652  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180824 05:59:34.585582 176306 server/server.go:1807  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180824 05:59:34.593321 176306 server/server.go:1538  [n1] starting https server at 127.0.0.1:46695 (use: 127.0.0.1:46695)
I180824 05:59:34.593615 176306 server/server.go:1540  [n1] starting grpc/postgres server at 127.0.0.1:38525
I180824 05:59:34.593680 176306 server/server.go:1541  [n1] advertising CockroachDB node at 127.0.0.1:38525
I180824 05:59:34.606029 176585 server/status/recorder.go:652  [n1,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180824 05:59:34.616737 176566 storage/replica_command.go:289  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180824 05:59:34.935453 176612 storage/replica_command.go:289  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180824 05:59:35.254908 176174 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180824 05:59:35.294742 176319 storage/replica_command.go:289  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180824 05:59:35.680462 176588 storage/replica_command.go:289  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
W180824 05:59:35.772609 176622 storage/intent_resolver.go:668  [n1,s1] failed to push during intent resolution: failed to push "sql txn" id=bc409928 key=/Table/SystemConfigSpan/Start rw=true pri=0.01272964 iso=SERIALIZABLE stat=PENDING epo=0 ts=1535090375.444665706,0 orig=1535090375.444665706,0 max=1535090375.444665706,0 wto=false rop=false seq=12
I180824 05:59:35.950650 176620 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:2.0-12 User:root}
I180824 05:59:35.955512 176411 rpc/nodedialer/nodedialer.go:92  [consistencyChecker,n1,s1,r1/1:/{Min-System/}] connection to n1 established
I180824 05:59:36.086896 176558 storage/replica_command.go:289  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180824 05:59:36.547520 176624 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180824 05:59:36.586728 176660 storage/replica_command.go:289  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180824 05:59:36.741235 176695 storage/replica_command.go:289  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180824 05:59:36.927777 176668 storage/replica_command.go:289  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180824 05:59:37.100845 176726 storage/replica_command.go:289  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180824 05:59:37.342011 176672 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:b2e89bf8-4a82-4ebd-9d05-d77f1d4f7da9 User:root}
I180824 05:59:37.418211 176743 storage/replica_command.go:289  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180824 05:59:37.669668 176689 storage/replica_command.go:289  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180824 05:59:37.792021 176636 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I180824 05:59:37.993382 176733 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180824 05:59:37.999269 176775 storage/replica_command.go:289  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180824 05:59:38.155266 176306 server/server.go:1594  [n1] done ensuring all necessary migrations have run
I180824 05:59:38.155656 176306 server/server.go:1597  [n1] serving sql connections
I180824 05:59:38.288241 176658 storage/replica_command.go:289  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180824 05:59:38.355638 176777 server/server_update.go:67  [n1] no need to upgrade, cluster already at the newest version
I180824 05:59:38.454655 176779 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:38525} Attrs: Locality: ServerVersion:2.0-12 BuildTag:v2.1.0-alpha.20180702-1973-gd6a4a0d StartedAt:1535090374493993901 LocalityAddress:[]} ClusterID:778669e8-bbea-4418-8b9c-cc57e630d855 StartedAt:1535090374493993901 LastUp:1535090374493993901}
I180824 05:59:38.488479 176793 storage/replica_command.go:289  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180824 05:59:38.685307 176760 storage/replica_command.go:289  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180824 05:59:38.817121 176786 storage/replica_command.go:289  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180824 05:59:38.981853 176837 storage/replica_command.go:289  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180824 05:59:39.228014 176864 storage/replica_command.go:289  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180824 05:59:39.465551 176828 storage/replica_command.go:289  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180824 05:59:39.767373 176890 storage/replica_command.go:289  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180824 05:59:39.895731 176931 storage/replica_command.go:289  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180824 05:59:41.028161 176933 storage/replica_command.go:289  [split,n1,s1,r22/1:/{Table/51-Max}] initiating a split of this range at key /Table/52 [r23]
I180824 05:59:41.034373 176953 sql/event_log.go:126  [n1,client=127.0.0.1:47040,user=root] Event: "create_database", target: 52, info: {DatabaseName:t Statement:CREATE DATABASE t User:root}
I180824 05:59:41.208302 176953 sql/event_log.go:126  [n1,client=127.0.0.1:47040,user=root] Event: "create_table", target: 53, info: {TableName:t.public.test Statement:CREATE TABLE t.public.test (k INT PRIMARY KEY, v STRING) User:root}
==================
WARNING: DATA RACE
Write at 0x00c420702048 by goroutine 549:
  github.com/cockroachdb/cockroach/pkg/sql_test.(*TxnAborter).executorKnobs.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/txn_restart_test.go:376 +0xd3
  github.com/cockroachdb/cockroach/pkg/sql_test.TestErrorOnCommitFinalizesTxn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/txn_restart_test.go:1099 +0x63a
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Previous read at 0x00c420702048 by goroutine 926:
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState.func4()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:181 +0xa4
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:433 +0xe78
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1135 +0x1ed4
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:210 +0x97

Goroutine 549 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:978 +0x2cd
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:36 +0x15d
  main.main()
      _testmain.go:508 +0x22a

Goroutine 926 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:199 +0x49a
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:449 +0x54d
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:292 +0x1d1
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).insertRangeLogEvent()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/log.go:99 +0x38f
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).logSplit()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/log.go:121 +0x353
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).adminSplitWithDescriptor.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:329 +0x422
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:585 +0x5e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:692 +0xf6
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:584 +0x13b
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).adminSplitWithDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:292 +0xcc7
  github.com/cockroachdb/cockroach/pkg/storage.(*splitQueue).processAttempt()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/split_queue.go:149 +0x1c8
  github.com/cockroachdb/cockroach/pkg/storage.(*splitQueue).process()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/split_queue.go:127 +0x93
  github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processReplica()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:744 +0x313
  github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1.2()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:627 +0xdd
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:324 +0xf3
==================
--- FAIL: TestErrorOnCommitFinalizesTxn (7.99s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/7048f0ed75fd2dc42779cc98af2b94cafe9a6ec5

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestErrorOnCommitFinalizesTxn PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=920479&tab=buildLog

```
  main.main()
      _testmain.go:516 +0x22a
==================
I180922 06:29:35.890281 158131 sql/txn_restart_test.go:345  [n1,split,s1,r22/1:/{Table/51-Max},intExec=log-range-event] statement filter running on: INSERT INTO system.public.rangelog("timestamp", "rangeID", "storeID", "eventType", "otherRangeID", info) VALUES ($1, $2, $3, $4, $5, $6), with err=<nil>
I180922 06:29:35.947107 158014 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:39478,user=root,intExec=lease-insert] statement filter running on: INSERT INTO system.public.lease("descID", version, "nodeID", expiration) VALUES (53, 1, 1, '2018-09-22 06:34:52.243012+00:00'), with err=<nil>
I180922 06:29:35.976124 157711 storage/split_queue.go:116  [n1,split,s1,r22/1:/Table/5{1-2}] shouldQueue: shouldQ=false priority=0.0
I180922 06:29:35.976662 157711 storage/split_queue.go:116  [n1,split,s1,r23/1:/{Table/52-Max}] shouldQueue: shouldQ=true priority=1.0
I180922 06:29:35.981773 158115 storage/split_queue.go:170  [n1,split,s1,r22/1:/Table/5{1-2}] split done
I180922 06:29:35.982433 158115 storage/split_queue.go:116  [n1,split,s1,r22/1:/Table/5{1-2}] shouldQueue: shouldQ=false priority=0.0
I180922 06:29:35.983757 158015 storage/replica_command.go:298  [n1,split,s1,r23/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r24]
I180922 06:29:36.018387 157970 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:39478,user=root] statement filter running on: INSERT INTO t.public.test(k, v) VALUES (0, 'boulanger'), with err=<nil>
I180922 06:29:36.018790 157970 sql/txn_restart_test.go:355  [n1,client=127.0.0.1:39478,user=root] TxnAborter aborting txn for statement "INSERT INTO t.public.test(k, v) VALUES (0, 'boulanger')"
I180922 06:29:36.038148 158089 sql/txn_restart_test.go:345  [n1,split,s1,r23/1:/{Table/52-Max},intExec=log-range-event] statement filter running on: INSERT INTO system.public.rangelog("timestamp", "rangeID", "storeID", "eventType", "otherRangeID", info) VALUES ($1, $2, $3, $4, $5, $6), with err=<nil>
I180922 06:29:36.082160 157712 storage/split_queue.go:116  [n1,split,s1,r23/1:/Table/5{2-3}] shouldQueue: shouldQ=false priority=0.0
I180922 06:29:36.082568 157712 storage/split_queue.go:116  [n1,split,s1,r24/1:/{Table/53-Max}] shouldQueue: shouldQ=false priority=0.0
I180922 06:29:36.086840 158015 storage/split_queue.go:170  [n1,split,s1,r23/1:/Table/5{2-3}] split done
I180922 06:29:36.196361 158103 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:39482,user=root] statement filter running on: SET TRANSACTION PRIORITY HIGH, with err=<nil>
I180922 06:29:36.262858 158103 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:39482,user=root] statement filter running on: DELETE FROM t.public.test WHERE k = $1, with err=<nil>
I180922 06:29:36.270142 158103 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:39482,user=root] statement filter running on: COMMIT TRANSACTION, with err=<nil>
I180922 06:29:36.279915 157970 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:39478,user=root] statement filter running on: COMMIT TRANSACTION, with err=HandledRetryableTxnError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" id=0c3654b3 key=/Table/53/1/0/0 rw=true pri=99.99999995 iso=SERIALIZABLE stat=ABORTED epo=0 ts=1537597775.900963632,0 orig=1537597775.900963632,0 max=1537597775.900963632,0 wto=false rop=false seq=3
I180922 06:29:36.301817 158075 sql/txn_restart_test.go:345  [intExec=get-table-statistics] statement filter running on: SELECT "tableID", "statisticID", name, "columnIDs", "createdAt", "rowCount", "distinctCount", "nullCount", histogram FROM system.public.table_statistics WHERE "tableID" = $1 ORDER BY "createdAt" DESC, with err=<nil>
I180922 06:29:36.307820 157970 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:39478,user=root] statement filter running on: SELECT * FROM t.public.test, with err=<nil>
I180922 06:29:36.310078 157970 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:39478,user=root] statement filter running on: COMMIT TRANSACTION, with err=<nil>
    --- FAIL: TestErrorOnCommitFinalizesTxn/retryIntent=false (0.43s)
    	testing.go:730: race detected during execution of test

```

SHA: https://github.com/cockroachdb/cockroach/commits/18519949df8e717c31a636bb153e526f45375ca8

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestErrorOnCommitFinalizesTxn PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=920442&tab=buildLog

```
I180922 08:56:15.555387 162030 storage/split_queue.go:116  [n1,split,s1,r20/1:/Table/{23-50}] shouldQueue: shouldQ=false priority=0.0
I180922 08:56:15.556489 162030 storage/split_queue.go:116  [n1,split,s1,r1/1:/{Min-System/}] shouldQueue: shouldQ=false priority=0.0
I180922 08:56:15.557186 162030 storage/split_queue.go:116  [n1,split,s1,r6/1:/{System/tse-Table/System…}] shouldQueue: shouldQ=false priority=0.0
I180922 08:56:15.626304 161978 storage/split_queue.go:116  [n1,split,s1,r22/1:/Table/5{1-2}] shouldQueue: shouldQ=false priority=0.0
I180922 08:56:15.626765 161978 storage/split_queue.go:116  [n1,split,s1,r23/1:/{Table/52-Max}] shouldQueue: shouldQ=true priority=1.0
I180922 08:56:15.634795 162357 storage/replica_command.go:298  [n1,split,s1,r23/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r24]
I180922 08:56:15.638013 162354 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:44538,user=root,intExec=lease-insert] statement filter running on: INSERT INTO system.public.lease("descID", version, "nodeID", expiration) VALUES (53, 1, 1, '2018-09-22 09:02:11.912153+00:00'), with err=<nil>
I180922 08:56:15.641698 162305 storage/split_queue.go:170  [n1,split,s1,r22/1:/Table/5{1-2}] split done
I180922 08:56:15.642490 162305 storage/split_queue.go:116  [n1,split,s1,r22/1:/Table/5{1-2}] shouldQueue: shouldQ=false priority=0.0
I180922 08:56:15.666673 162246 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:44538,user=root] statement filter running on: INSERT INTO t.public.test(k, v) VALUES (0, 'boulanger'), with err=<nil>
I180922 08:56:15.666995 162246 sql/txn_restart_test.go:355  [n1,client=127.0.0.1:44538,user=root] TxnAborter aborting txn for statement "INSERT INTO t.public.test(k, v) VALUES (0, 'boulanger')"
I180922 08:56:15.675343 162348 sql/txn_restart_test.go:345  [n1,split,s1,r23/1:/{Table/52-Max},intExec=log-range-event] statement filter running on: INSERT INTO system.public.rangelog("timestamp", "rangeID", "storeID", "eventType", "otherRangeID", info) VALUES ($1, $2, $3, $4, $5, $6), with err=<nil>
I180922 08:56:15.699891 161987 storage/split_queue.go:116  [n1,split,s1,r23/1:/Table/5{2-3}] shouldQueue: shouldQ=false priority=0.0
I180922 08:56:15.700408 161987 storage/split_queue.go:116  [n1,split,s1,r24/1:/{Table/53-Max}] shouldQueue: shouldQ=false priority=0.0
I180922 08:56:15.704149 162357 storage/split_queue.go:170  [n1,split,s1,r23/1:/Table/5{2-3}] split done
I180922 08:56:15.777428 162387 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:44540,user=root] statement filter running on: SET TRANSACTION PRIORITY HIGH, with err=<nil>
I180922 08:56:15.816107 162387 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:44540,user=root] statement filter running on: DELETE FROM t.public.test WHERE k = $1, with err=<nil>
I180922 08:56:15.826293 162387 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:44540,user=root] statement filter running on: COMMIT TRANSACTION, with err=<nil>
I180922 08:56:15.831625 162246 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:44538,user=root] statement filter running on: COMMIT TRANSACTION, with err=HandledRetryableTxnError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" id=893bfb8b key=/Table/53/1/0/0 rw=true pri=99.99999995 iso=SERIALIZABLE stat=ABORTED epo=0 ts=1537606575.541817299,0 orig=1537606575.541817299,0 max=1537606575.541817299,0 wto=false rop=false seq=3
I180922 08:56:15.862260 162351 sql/txn_restart_test.go:345  [intExec=get-table-statistics] statement filter running on: SELECT "tableID", "statisticID", name, "columnIDs", "createdAt", "rowCount", "distinctCount", "nullCount", histogram FROM system.public.table_statistics WHERE "tableID" = $1 ORDER BY "createdAt" DESC, with err=<nil>
I180922 08:56:15.867997 162246 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:44538,user=root] statement filter running on: SELECT * FROM t.public.test, with err=<nil>
I180922 08:56:15.870795 162246 sql/txn_restart_test.go:345  [n1,client=127.0.0.1:44538,user=root] statement filter running on: COMMIT TRANSACTION, with err=<nil>
    --- FAIL: TestErrorOnCommitFinalizesTxn/retryIntent=false (0.34s)
    	testing.go:730: race detected during execution of test

```

SHA: https://github.com/cockroachdb/cockroach/commits/304150d8010b05cd5bd106e6e6b63a89c2835403

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestErrorOnCommitFinalizesTxn PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=921224&tab=buildLog

```
  github.com/cockroachdb/cockroach/pkg/sql_test.TestErrorOnCommitFinalizesTxn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/txn_restart_test.go:1099 +0x63a
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Previous read at 0x00c42086a008 by goroutine 1599:
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState.func4()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:181 +0xa4
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:435 +0xe43
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x457
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1136 +0x1f66
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:211 +0x97

Goroutine 445 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:978 +0x2cd
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:36 +0x15d
  main.main()
      _testmain.go:528 +0x22a

Goroutine 1599 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:200 +0x492
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:450 +0x54d
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:293 +0x1d1
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).insertRangeLogEvent()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/log.go:99 +0x38f
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).logSplit()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/log.go:121 +0x353
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).adminSplitWithDescriptor.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:338 +0x422
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:586 +0x5e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:705 +0xf6
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:585 +0x15d
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).adminSplitWithDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:301 +0xcc7
  github.com/cockroachdb/cockroach/pkg/storage.(*splitQueue).processAttempt()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/split_queue.go:154 +0x1ae
  github.com/cockroachdb/cockroach/pkg/storage.(*splitQueue).process()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/split_queue.go:132 +0x70
  github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processReplica()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:742 +0x2e3
  github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1.2()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:626 +0xdd
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:324 +0xf3
==================
--- FAIL: TestErrorOnCommitFinalizesTxn (3.80s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/b2bd8e8b6446a566b667be6094e019c1040ed98d

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestErrorOnCommitFinalizesTxn PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=924209&tab=buildLog

```
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Previous read at 0x00c42173a008 by goroutine 645:
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState.func4()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:181 +0xa4
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:435 +0xe43
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x457
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1137 +0x1f66
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:211 +0x97

Goroutine 1098 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:978 +0x2cd
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:36 +0x15d
  main.main()
      _testmain.go:528 +0x22a

Goroutine 645 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:200 +0x492
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:450 +0x54d
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:293 +0x1d1
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).insertRangeLogEvent()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/log.go:99 +0x38f
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).logSplit()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/log.go:121 +0x353
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).adminSplitWithDescriptor.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:338 +0x422
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:586 +0x5e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:705 +0xf6
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:585 +0x15d
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).adminSplitWithDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:301 +0xcc7
  github.com/cockroachdb/cockroach/pkg/storage.(*splitQueue).processAttempt()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/split_queue.go:154 +0x1ae
  github.com/cockroachdb/cockroach/pkg/storage.(*splitQueue).process()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/split_queue.go:132 +0x70
  github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processReplica()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:742 +0x2e3
  github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1.2()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:626 +0xdd
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:324 +0xf3
==================
I180925 06:58:46.671969 161616 storage/split_queue.go:116  [n1,split,s1,r5/1:/System/ts{d-e}] shouldQueue: shouldQ=false priority=0.0
--- FAIL: TestErrorOnCommitFinalizesTxn (4.18s)

```

--- a/pkg/sql/conn_executor_test.go
+++ b/pkg/sql/conn_executor_test.go
@@ -93,8 +93,7 @@ func TestSessionFinishRollsBackTxn(t *testing.T) {
 	aborter := NewTxnAborter()
 	defer aborter.Close(t)
 	params, _ := tests.CreateTestServerParams()
-	var activateKnobs func()
-	params.Knobs.SQLExecutor, activateKnobs = aborter.executorKnobs()
+	params.Knobs.SQLExecutor = aborter.executorKnobs()
 	s, mainDB, _ := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(context.TODO())
 	{
@@ -106,15 +105,9 @@ func TestSessionFinishRollsBackTxn(t *testing.T) {
 		}
 	}
 
-	// After this point, the abort knob is active. This also means that
-	// the AST is checked for modification. So we have to use fully
-	// qualified table names throughout to avoid a mismatch due to table
-	// qualification.
-	activateKnobs()
-
 	if _, err := mainDB.Exec(`
 CREATE DATABASE t;
-CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT);
+CREATE TABLE t.test (k INT PRIMARY KEY, v TEXT);
 `); err != nil {
 		t.Fatal(err)
 	}
@@ -217,7 +210,7 @@ CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT);
 			}
 			ts := timeutil.Now()
 			var count int
-			if err := txCheck.QueryRow("SELECT count(1) FROM t.public.test").Scan(&count); err != nil {
+			if err := txCheck.QueryRow("SELECT count(1) FROM t.test").Scan(&count); err != nil {
 				t.Fatal(err)
 			}
 			// CommitWait actually committed, so we'll need to clean up.
@@ -226,7 +219,7 @@ CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT);
 					t.Fatalf("expected no rows, got: %d", count)
 				}
 			} else {
-				if _, err := txCheck.Exec("DELETE FROM t.public.test"); err != nil {
+				if _, err := txCheck.Exec("DELETE FROM t.test"); err != nil {
 					t.Fatal(err)
 				}
 			}

--- a/pkg/sql/txn_restart_test.go
+++ b/pkg/sql/txn_restart_test.go
@@ -278,8 +278,13 @@ var valuesRE = regexp.MustCompile(`VALUES.*\((\d),`)
 
 // QueueStmtForAbortion registers a statement whose transaction will be aborted.
 //
-// stmt needs to be the statement, literally as the parser will convert it back
-// to a string.
+// stmt needs to be the statement, literally as the AST gets converted back to a
+// string. Note that, since we sometimes change the AST during planning, the
+// statements sent for execution that need to be intercepted by this filter
+// need to be written in a canonical form, and stmt passed here needs to also be
+// that canonical form. In particular, table names need to be fully qualified
+// with the schema (e.g. t.public.test).
+//
 // abortCount specifies how many times a txn running this statement will be
 // aborted.
 // willBeRetriedIbid should be set if the statement will be retried by the test
@@ -366,14 +371,12 @@ func (ta *TxnAborter) statementFilter(ctx context.Context, stmt string, err erro
 }
 
 // executorKnobs are the bridge between the TxnAborter and the sql.Executor.
-// The aborter is only set up when the 2nd return value is called.
-func (ta *TxnAborter) executorKnobs() (base.ModuleTestingKnobs, func()) {
-	knobs := &sql.ExecutorTestingKnobs{}
-	return knobs, func() {
+func (ta *TxnAborter) executorKnobs() base.ModuleTestingKnobs {
+	return &sql.ExecutorTestingKnobs{
 		// We're going to abort txns using a TxnAborter, and that's incompatible
 		// with AutoCommit.
-		knobs.DisableAutoCommit = true
-		knobs.StatementFilter = ta.statementFilter
+		DisableAutoCommit: true,
+		StatementFilter:   ta.statementFilter,
 	}
 }
 
@@ -450,8 +453,7 @@ func TestTxnAutoRetry(t *testing.T) {
 	aborter := NewTxnAborter()
 	defer aborter.Close(t)
 	params, cmdFilters := tests.CreateTestServerParams()
-	var activateKnobs func()
-	params.Knobs.SQLExecutor, activateKnobs = aborter.executorKnobs()
+	params.Knobs.SQLExecutor = aborter.executorKnobs()
 	s, sqlDB, _ := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(context.TODO())
 	{
@@ -472,15 +474,9 @@ func TestTxnAutoRetry(t *testing.T) {
 	// do that.
 	sqlDB.SetMaxOpenConns(1)
 
-	// After this point, the abort knob is active. This also means that
-	// the AST is checked for modification. So we have to use fully
-	// qualified table names throughout to avoid a mismatch due to table
-	// qualification.
-	activateKnobs()
-
 	if _, err := sqlDB.Exec(`
 CREATE DATABASE t;
-CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT, t DECIMAL);
+CREATE TABLE t.test (k INT PRIMARY KEY, v TEXT, t DECIMAL);
 `); err != nil {
 		t.Fatal(err)
 	}
@@ -788,8 +784,7 @@ func TestAbortedTxnOnlyRetriedOnce(t *testing.T) {
 	aborter := NewTxnAborter()
 	defer aborter.Close(t)
 	params, _ := tests.CreateTestServerParams()
-	var activateKnobs func()
-	params.Knobs.SQLExecutor, activateKnobs = aborter.executorKnobs()
+	params.Knobs.SQLExecutor = aborter.executorKnobs()
 	s, sqlDB, _ := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(context.TODO())
 	{
@@ -800,12 +795,6 @@ func TestAbortedTxnOnlyRetriedOnce(t *testing.T) {
 		}
 	}
 
-	// After this point, the abort knob is active. This also means that
-	// the AST is checked for modification. So we have to use fully
-	// qualified table names throughout to avoid a mismatch due to table
-	// qualification.
-	activateKnobs()
-
 	const insertStmt = "INSERT INTO t.public.test(k, v) VALUES (1, 'boulanger')"
 	if err := aborter.QueueStmtForAbortion(
 		insertStmt, 1 /* abortCount */, true, /* willBeRetriedIbid */
@@ -815,7 +804,7 @@ func TestAbortedTxnOnlyRetriedOnce(t *testing.T) {
 
 	if _, err := sqlDB.Exec(`
 CREATE DATABASE t;
-CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT);
+CREATE TABLE t.test (k INT PRIMARY KEY, v TEXT);
 `); err != nil {
 		t.Fatal(err)
 	}
@@ -950,8 +939,7 @@ func TestTxnUserRestart(t *testing.T) {
 				aborter := NewTxnAborter()
 				defer aborter.Close(t)
 				params, cmdFilters := tests.CreateTestServerParams()
-				var activateKnobs func()
-				params.Knobs.SQLExecutor, activateKnobs = aborter.executorKnobs()
+				params.Knobs.SQLExecutor = aborter.executorKnobs()
 				s, sqlDB, _ := serverutils.StartServer(t, params)
 				defer s.Stopper().Stop(context.TODO())
 				{
@@ -962,15 +950,9 @@ func TestTxnUserRestart(t *testing.T) {
 					}
 				}
 
-				// After this point, the abort knob is active. This also means that
-				// the AST is checked for modification. So we have to use fully
-				// qualified table names throughout to avoid a mismatch due to table
-				// qualification.
-				activateKnobs()
-
 				if _, err := sqlDB.Exec(`
 CREATE DATABASE t;
-CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT);
+CREATE TABLE t.test (k INT PRIMARY KEY, v TEXT);
 `); err != nil {
 					t.Fatal(err)
 				}
@@ -998,7 +980,7 @@ CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT);
 				checkRestarts(t, tc.magicVals)
 
 				// Check that we only wrote the sentinel row.
-				rows, err := sqlDB.Query("SELECT * FROM t.public.test")
+				rows, err := sqlDB.Query("SELECT * FROM t.test")
 				if err != nil {
 					t.Fatal(err)
 				}
@@ -1021,7 +1003,7 @@ CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT);
 					t.Error(err)
 				}
 				// Clean up the table for the next test iteration.
-				_, err = sqlDB.Exec("DELETE FROM t.public.test WHERE true")
+				_, err = sqlDB.Exec("DELETE FROM t.test WHERE true")
 				if err != nil {
 					t.Fatal(err)
 				}
@@ -1071,8 +1053,7 @@ func TestErrorOnCommitFinalizesTxn(t *testing.T) {
 	aborter := NewTxnAborter()
 	defer aborter.Close(t)
 	params, _ := tests.CreateTestServerParams()
-	var activateKnobs func()
-	params.Knobs.SQLExecutor, activateKnobs = aborter.executorKnobs()
+	params.Knobs.SQLExecutor = aborter.executorKnobs()
 	s, sqlDB, _ := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(context.TODO())
 	{
@@ -1091,12 +1072,6 @@ CREATE DATABASE t; CREATE TABLE t.test (k INT PRIMARY KEY, v TEXT);
 	// We need to do everything on one connection as we'll want to observe the
 	// connection state after a COMMIT.
 	sqlDB.SetMaxOpenConns(1)
-
-	// After this point, the abort knob is active. This also means that
-	// the AST is checked for modification. So we have to use fully
-	// qualified table names throughout to avoid a mismatch due to table
-	// qualification.
-	activateKnobs()
 
 	// We're going to test both errors that would leave the transaction in the
 	// RestartWait state and errors that would leave the transaction in Aborted,
@@ -1138,7 +1113,7 @@ CREATE DATABASE t; CREATE TABLE t.test (k INT PRIMARY KEY, v TEXT);
 				t.Fatal(err)
 			}
 			// Check that we don't see any rows, so the previous txn was rolled back.
-			rows, err := sqlDB.Query("SELECT * FROM t.public.test")
+			rows, err := sqlDB.Query("SELECT * FROM t.test")
 			if err != nil {
 				t.Fatal(err)
 			}
@@ -1164,8 +1139,7 @@ func TestRollbackInRestartWait(t *testing.T) {
 	aborter := NewTxnAborter()
 	defer aborter.Close(t)
 	params, _ := tests.CreateTestServerParams()
-	var activateKnobs func()
-	params.Knobs.SQLExecutor, activateKnobs = aborter.executorKnobs()
+	params.Knobs.SQLExecutor = aborter.executorKnobs()
 	s, sqlDB, _ := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(context.TODO())
 	{
@@ -1182,12 +1156,6 @@ CREATE TABLE t.test (k INT PRIMARY KEY, v TEXT);
 `); err != nil {
 		t.Fatal(err)
 	}
-
-	// After this point, the abort knob is active. This also means that
-	// the AST is checked for modification. So we have to use fully
-	// qualified table names throughout to avoid a mismatch due to table
-	// qualification.
-	activateKnobs()
 
 	// Set up error injection that causes retries.
 	const insertStmt = "INSERT INTO t.public.test(k, v) VALUES (0, 'boulanger')"

